### ISSUE TITLE
API Docs Enhancement: Public and Private URLs

### ISSUE THREAD

**Issue by mjones-oddball** [department-of-veterans-affairs/notification-api#1377]
## Business Need

Each time VA Notify onboards a client we have to manually explain the public and private URLs and guide them to use the appropriate one. Currently, our API docs only list the private URLs, though it does not explicitly state that they are private.

The private URL will only work if the triggering application is within the VA Network. We want clients to know the public URL exists, but the preference is that they should be using private unless their application is outside the VA Network.

- see suggestion from QA in the comments below in the event I am OOO

- [x] Discussed with QA

### User Story(ies)
**As a** VA business line integrating with VA Notify
**I want** to know which API URL is appropriate to use
**So that** I can successfully send notifications from my application

### Additional Info and Resources
- Public URLs below. Please note we only want clients using these when their application is outside the VA network. We should make the requirements clear to avoid extra traffic routing through public unnecessarily. 
    - https://dev-api.va.gov/vanotify
    - https://staging-api.va.gov/vanotify
    - https://api.va.gov/vanotify
- Current API Docs:
![image.png](https://images.zenhubusercontent.com/62041308731c2b2cdc47288b/6810520f-c4fd-4551-a11c-aca03488cd7d)

## Engineering Checklist
_**Required**_
- [x] Update [API documents](https://github.com/department-of-veterans-affairs/notification-api/tree/master/documents/openapi) to include both private and public URLs as well as an explanation of when to use each
    - [x] Add perf as an available environment in the computed URL section for both private and public URL
    - [x] Get Kyle & Melanie's approval on content
- [x] Ping Strike team when this change is complete/merged so they can run the script that pulls in the api docs to portal at their next deployment

## Acceptance Criteria
_**Required**_

GIVEN a technical user wants to trigger a notification from an application hosted behind the VA Network
WHEN they navigate to the API documentation in the Self Service UI
THEN they can identify the right type of URL to use for their desired environment

GIVEN a technical user wants to trigger a notification from an application which is hosted outside the VA Network
WHEN they navigate to the API documentation in the Self Service UI
THEN they can identify the right type of URL to use for their desired environment

## QA Considerations
_For QA to populate. Leave blank if QA is not applicable on this ticket._
- Works in swagger (since we aren't validating Portal) and is clear on which URL to use in which circumstance

## Potential Dependencies
_Leave blank if n/a_

## Out of Scope
_Leave blank if n/a_
- Portal Deployment
- Validating Portal

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1377?planning-poker) with Zenhub @cris-oddball @EvanParish @justaskdavidb2 @k-macmillan @kalbfled @ldraney @nikolai-efimov

**Comment by tabinda-syed** [comment]
Team will ping Strike to give a heads-up once this ticket is done.

**Comment by cris-oddball** [comment]
Suggestions for whomever picks up this ticket; there are at least two ways to go here.
[Docs](https://swagger.io/specification/v3/#server-object)

1. Declare everything in an array
```
servers:
  - url: 'https://api.notifications.va.gov'
    description: Production VPC endpoint
  - url: 'https://api.staging.notifications.va.gov'
    description: Staging VPC endpoint
  - url: 'https://perf.notifications.va.gov'
    description: Perf VPC Endpoint
  - url: 'https://dev.notifications.va.gov'
    description: Dev VPC Endpoint
  - url: 'https://api.va.gov/vanotify'
    description: Production Public Endpoint
  - url: 'https://staging-api.va.gov/vanotify'
    description: Staging Public endpoint
  - url: 'https://sandbox-api.va.gov/vanotify'
    description: Perf Public Endpoint
  - url: 'https://dev-api.va.gov/vanotify'
    description: Dev VPC Endpoint
```
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMWZwQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--4c37c5fd40b787ffb83b5232f40db247f3288f99/Screenshot%202023-11-17%20at%204.00.05%E2%80%AFPM.jpg)
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMWpwQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--8094ec67049141afc428fd286684380e0ce8966a/Screenshot%202023-11-17%20at%203.59.46%E2%80%AFPM.jpg)

2.  Use enums and descriptions (probably more aesthetically pleasing)
```
servers:
  - url: 'https://{environment}.notifications.va.gov'
    description: VPC endpoints; must be on the VA network
    variables:
      environment:
        enum:
          - api
          - staging-api
          - perf-api
          - dev.api
        default: api
  - url: 'https://{environment}.va.gov/vanotify'
    description: Public Endpoints for use outside of the VPC
    variables:
      environment:
        enum:
          - api
          - staging-api
          - sandbox-api
          - dev-api
        default: api
```

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMXJwQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--4b58aaa44b7bb91b3b881b635eff74e921e6a532/Screenshot%202023-11-17%20at%204.01.30%E2%80%AFPM.jpg)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMXZwQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--4c627e05db49ebef990a0fa5bd5cf178298dd09b/Screenshot%202023-11-17%20at%204.01.44%E2%80%AFPM.jpg)

**Comment by cris-oddball** [comment]
Made modifications based on Kyle's feedback, pinged Strike, closing this ticket!
